backend/src/test/java/com/github/redstylzz/backend/service/CategoryServiceTest.java

```diff
@@ -78,20 +81,44 @@ void shouldReturnListOnSuccessfulRename() {
         when(repository.findByUserIDAndCategoryID(anyString(), anyString())).thenReturn(TestDataProvider.testCategory());
         when(repository.existsByUserIDAndCategoryName(anyString(), anyString())).thenReturn(false);
 
-        underTest.renameCategory(user, "", "");
+        List<Category> wantedList = underTest.renameCategory(user, "", "");
 
         verify(repository).save(any(Category.class));
         verify(repository).findAllByUserID(anyString());
-
+        assertEquals(List.of(), wantedList);
     }
 
     @Test
     void shouldReturnListOnSuccessfulDelete() {
         MongoUser user = TestDataProvider.testUser();
 
-        underTest.deleteCategory(user, "");
+        List<Category> wantedList = underTest.deleteCategory(user, "");
 
         verify(repository).deleteByCategoryID(anyString());
+        verify(paymentRepo).deleteAllByUserIDAndCategoryID(anyString(), anyString());
         verify(repository).findAllByUserID(anyString());
+        assertEquals(List.of(), wantedList);
```

Comment:
if you added a return in your method to be able to test is, using a verify to ensure an interaction with your repository is sufficient. In that case you should include your expected object within the verify statement